provisioning/device/src/Utilities/X509CertificateStore.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Utilities
+{
+    internal class X509CertificateStore : ICertificiateStore, IDisposable
```

Comment:
Maybe instead of an interface, we make the class methods virtual and a unit test just overrides them. How would that look in a unit test? That would rid us of the interface definition.